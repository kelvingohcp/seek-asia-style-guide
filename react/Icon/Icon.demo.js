import React from 'react';
import Icon from './Icon';
import IconPack from './Icon.path';
import PropTypes from 'prop-types';
import * as sketch from './Icon.sketch';
import { PageBlock } from 'seek-asia-style-guide/react';

const IconContainer = ({ component: DemoComponent, componentProps }) => {
  return (
    <PageBlock style={'width: 100%;'}>
      <DemoComponent {...componentProps} />
    </PageBlock>
  );
};

IconContainer.propTypes = {
  component: PropTypes.func,
  componentProps: PropTypes.object
};

const getSvgPaths = IconPack.map(obj => {
  const rObj = {
    label: obj.label,
    transformProps: props => ({
      ...props,
      type: obj.name
    })
  };
  return rObj;
});
export default {
  route: '/icon',
  title: 'Icon',
  component: Icon,
  sketch,
  initialProps: {
    type: 'birthdate',
    size: '',
    animation: ''
  },
  options: [
    {
      label: 'Type',
      type: 'radio',
      states: getSvgPaths
    },
    {
      label: 'Size',
      type: 'radio',
      states: [
        {
          label: 'Size unset',
          transformProps: props => props
        },
        {
          label: 'Small',
          transformProps: props => ({
            ...props,
            size: 'small'
          })
        },
        {
          label: 'Normal',
          transformProps: props => ({
            ...props,
            size: 'normal'
          })
        },
        {
          label: 'Large',
          transformProps: props => ({
            ...props,
            size: 'large'
          })
        }
      ]
    },
    {
      label: 'Line height',
      type: 'radio',
      states: [
        {
          label: 'Line height',
          transformProps: props => props
        },
        {
          label: 'Conversational',
          transformProps: props => ({
            ...props,
            lineHeight: 'conversational'
          })
        },
        {
          label: 'Intimate',
          transformProps: props => ({
            ...props,
            lineHeight: 'intimate'
          })
        },
        {
          label: 'Loud',
          transformProps: props => ({
            ...props,
            lineHeight: 'loud'
          })
        },
        {
          label: 'Screaming',
          transformProps: props => ({
            ...props,
            lineHeight: 'screaming'
          })
        },
        {
          label: 'Shouting',
          transformProps: props => ({
            ...props,
            lineHeight: 'shouting'
          })
        },
        {
          label: 'Waving',
          transformProps: props => ({
            ...props,
            lineHeight: 'waving'
          })
        },
        {
          label: 'Whispering',
          transformProps: props => ({
            ...props,
            lineHeight: 'whispering'
          })
        },
        {
          label: 'Whistling',
          transformProps: props => ({
            ...props,
            lineHeight: 'whistling'
          })
        },
        {
          label: 'Yelling',
          transformProps: props => ({
            ...props,
            lineHeight: 'yelling'
          })
        }
      ]
    },
    {
      label: 'Rotate',
      type: 'radio',
      states: [
        {
          label: 'No rotation',
          transformProps: props => props
        },
        {
          label: '90 degree',
          transformProps: props => ({
            ...props,
            rotation: '90deg'
          })
        },
        {
          label: '180 degree',
          transformProps: props => ({
            ...props,
            rotation: '180deg'
          })
        },
        {
          label: '-90 degree',
          transformProps: props => ({
            ...props,
            rotation: '-90deg'
          })
        },
        {
          label: '-180 degree',
          transformProps: props => ({
            ...props,
            rotation: '-180deg'
          })
        },
        {
          label: '90 deg (animated)',
          transformProps: props => ({
            ...props,
            smoothRotate: true,
            rotation: '90deg'
          })
        },
        {
          label: '180 deg (animated)',
          transformProps: props => ({
            ...props,
            smoothRotate: true,
            rotation: '180deg'
          })
        },
        {
          label: '-90 deg (animated)',
          transformProps: props => ({
            ...props,
            smoothRotate: true,
            rotation: '-90deg'
          })
        },
        {
          label: '-180 deg (animated)',
          transformProps: props => ({
            ...props,
            smoothRotate: true,
            rotation: '-180deg'
          })
        }
      ]
    },
    {
      label: 'One time animation',
      type: 'radio',
      states: [
        {
          label: 'One time animation',
          transformProps: props => props
        },
        {
          label: 'Bounce',
          transformProps: props => ({
            ...props,
            animation: 'bounce'
          })
        },
        {
          label: 'Pop in',
          transformProps: props => ({
            ...props,
            animation: 'popIn'
          })
        }
      ]
    }
  ]
};
