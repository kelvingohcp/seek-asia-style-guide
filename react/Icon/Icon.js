import React from 'react';
import PropTypes from 'prop-types';
import IconPack from './Icon.path';
import styles from './Icon.less';
import classnames from 'classnames';

export default function Icon({
  animation,
  className,
  lineHeight,
  rotation,
  size,
  smoothRotate,
  title,
  type,
  svgClassName,
  ...restProps
}) {
  const iconData = IconPack.find(x => x.name === type);

  return (
    <span
      className={classnames({
        [styles.root]: true,
        [styles.sizeSmall]: size === 'small',
        [styles.sizeNormal]: size === 'normal',
        [styles.sizeLarge]: size === 'large',
        [styles.smallBounce]: animation === 'bounce',
        [styles.popIn]: animation === 'popIn',
        [styles.conversational]: lineHeight === 'conversational',
        [styles.lineHeightIntimate]: lineHeight === 'intimate',
        [styles.lineHeightLoud]: lineHeight === 'loud',
        [styles.lineHeightScreaming]: lineHeight === 'screaming',
        [styles.lineHeightShouting]: lineHeight === 'shouting',
        [styles.lineHeightWaving]: lineHeight === 'waving',
        [styles.lineHeightWhispering]: lineHeight === 'whispering',
        [styles.lineHeightWhistling]: lineHeight === 'whistling',
        [styles.lineHeightYelling]: lineHeight === 'yelling',
        [className]: className
      })} role="presentation">
      <svg
        xmlns={'http://www.w3.org/2000/svg'}
        aria-hidden={'true'}
        focusable={'false'}
        {...restProps}
        className={classnames({
          [svgClassName]: svgClassName,
          [styles.rotate90]: rotation === '90deg',
          [styles.rotate90cc]: rotation === '-90deg',
          [styles.rotate180]: rotation === '180deg',
          [styles.rotate180cc]: rotation === '-180deg',
          [styles.rotateReset]: rotation === 'reset',
          [styles.smoothRotate]: smoothRotate
        })} viewBox="0 0 100 100" role="img">
        <title>{title ? title : iconData.label}</title>
        <path d={iconData.path} />
      </svg>
    </span>
  );
}

Icon.propTypes = {
  animation: PropTypes.oneOf(['', 'bounce', 'popIn']),
  className: PropTypes.string,
  svgClassName: PropTypes.string,
  lineHeight: PropTypes.oneOf(['', 'conversational', 'intimate', 'loud', 'screaming', 'shouting', 'waving', 'whispering', 'whistling', 'yelling']),
  rotation: PropTypes.oneOf(['', '90deg', '-90deg', '180deg', '-180deg', 'reset']),
  size: PropTypes.oneOf(['', 'large', 'normal', 'small']),
  smoothRotate: PropTypes.bool,
  title: PropTypes.string,
  type: PropTypes.oneOf([
    'add',
    'alert',
    'android',
    'appDownload',
    'apple',
    'application',
    'applicationHistory',
    'birthdate',
    'bonus',
    'bookmark',
    'calendar',
    'careerInsight',
    'careerLevel',
    'certifications',
    'check',
    'chevron',
    'chime',
    'comments',
    'company',
    'companyProfile',
    'country',
    'currentStatus',
    'dentalInsurance',
    'doublePay',
    'download',
    'edit',
    'education',
    'employer',
    'employmentType',
    'exit',
    'expand',
    'experience',
    'expired',
    'extend',
    'facebook',
    'featured',
    'fiveDayWork',
    'flexibleWorkingHours',
    'freeShuttleBus',
    'gratuity',
    'hidden',
    'home',
    'housingAllowance',
    'information',
    'instagram',
    'jobAd',
    'jobAlert',
    'jobFunction',
    'jobInvitation',
    'language',
    'lifeInsurance',
    'location',
    'locked',
    'mail',
    'medical',
    'menu',
    'myJobsDB',
    'negative',
    'overtimePay',
    'pastApplication',
    'positive',
    'preferences',
    'profileFemale',
    'profileMale',
    'promotion',
    'purchaseCredits',
    'RC',
    'refresh',
    'remove',
    'resources',
    'resume',
    'rss',
    'salary',
    'search',
    'settings',
    'share',
    'show',
    'skills',
    'talentSearch',
    'telephone',
    'timestamp',
    'transportAllowance',
    'travelAllowance',
    'twitter',
    'unlocked',
    'upload',
    'youtube'
  ])
};

Icon.defaultProps = {
  animation: '',
  className: '',
  lineHeight: '',
  rotation: '',
  size: '',
  smoothRotate: false,
  type: ''
};
