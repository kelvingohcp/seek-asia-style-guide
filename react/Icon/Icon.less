@import (reference) '~seek-asia-style-guide/theme';

.root {
  display: inline-block;
  width: @grid-base * 6;
  height: @grid-base * 6;
  line-height: 1;
}

.small {
  width: @grid-base * 4;
  height: @grid-base * 4;
}

.large {
  width: @grid-base * 9;
  height: @grid-base * 9;
}

.smoothRotate {
  transition: transform 0.3s;
}

.rotate90 {
  transform: rotate(90deg);
}

.rotate180 {
  transform: rotate(180deg);
}

.rotate90cc {
  transform: rotate(-90deg);
}

.rotate180cc {
  transform: rotate(-180deg);
}

.smallBounce {
  animation: bounce 0.3s both;
}

@keyframes bounce {
  from,
  to {
    animation-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);
    transform: translate3d(0, 0, 0);
  }
  40%,
  50% {
    animation-timing-function: cubic-bezier(0.755, 0.05, 0.855, 0.06);
    transform: translate3d(0, -6px, 0);
  }
}

@keyframes rotate90 {
  to {
    transform: rotate(90deg);
  }
}

@keyframes rotate90cc {
  to {
    transform: rotate(-90deg);
  }
}

@keyframes rotate180 {
  to {
    transform: rotate(180deg);
  }
}

@keyframes rotate180cc {
  to {
    transform: rotate(-180deg);
  }
}

.popIn {
  animation: popIn 0.3s ease-in-out;
}

@keyframes popIn {
  0% { transform: scale(0.6); }
  50% { transform: scale(1.1); }
  100% { transform: scale(1); }
}
