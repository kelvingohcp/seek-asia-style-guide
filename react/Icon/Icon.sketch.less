@import (reference) '~seek-asia-style-guide/theme';

.demoInline {
  display: flex;
  align-items: flex-start;
  @supports (display: grid) {
    display: grid;
    justify-content: start;
    grid-auto-flow: column;
    grid-gap: @grid-base;
  }
}

.customizeIconStyle {
  color: goldenrod;
  width: 22px;
  height: 22px;
}

.iconGroup {
  display: flex;
  flex-wrap: wrap;
  @supports (display: grid) {
    display: grid;
    grid-column-gap: @grid-base * 4;
    grid-row-gap: @grid-base * 8;
    grid-template-columns: repeat(auto-fit, minmax(10em, 1fr));
  }
}

.iconGroupItem {
  flex-basis: 10em;
  font-size: @grid-base * 3 * 1px;
  @supports (display: grid) {
    display: grid;
    grid-auto-flow: row;
    grid-gap: @grid-base * 2;
    grid-auto-rows: (@grid-base * 6) 1fr;
  }
}
