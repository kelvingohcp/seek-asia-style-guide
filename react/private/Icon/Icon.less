@import (reference) "~seek-asia-style-guide/theme";

@iconWidth: @grid-base * 4;
@iconHeight: @grid-base * 4;

.root {
  display: inline-block;
}

.svg {
  fill: currentColor;
  width: @iconWidth;
  height: @iconHeight;
}
