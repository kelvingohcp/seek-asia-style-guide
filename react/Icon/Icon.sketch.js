import React from 'react';
import styles from './Icon.sketch.less';

import Button from '../Button/Button';
import Icon from './Icon';
import Text from '../Text/Text';

export const symbols = {
  'Job Alert, set size as normal': <Icon type="jobAlert" size="normal" />,
  'Chevron, 90 deg clockwise': <Icon rotation="90deg" type="chevron" />,
  'Job Ad, large': <Icon type="jobAd" size="large" />,
  'Bookmark, small': <Icon type="bookmark" size="small" />,
  'Bookmark icon wrapped with link': <a href="#"><Icon type="bookmark" /></a>,
  'Icon in front of text': <div className={styles.demoInline}><Icon type="promotion" size="normal" lineHeight="shouting" /> <Text baseline={false}>Promotion</Text></div>,
  'Icon respects font color value from parent container': <div><Button color="callToAction" isJobStreet><Icon type="resume" size="small" lineHeight="conversational" /> Resume</Button> <Button color="callToAction" compact><Icon type="resume" size="small" lineHeight="conversational" /> Resume</Button></div>
};
