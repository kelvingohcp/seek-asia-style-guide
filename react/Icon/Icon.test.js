import React from 'react';
import { shallow } from 'enzyme';

import Icon from './Icon';

describe('Icon', () => {
  const defaultProps = {
    type: 'chevron'
  };
  it('renders with default props', () => {
    const wrapper = shallow(<Icon {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with title', () => {
    const wrapper = shallow(<Icon title='title' {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the small style', () => {
    const wrapper = shallow(<Icon size="small" {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the large style', () => {
    const wrapper = shallow(<Icon size="large" {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the 90 deg rotated style', () => {
    const wrapper = shallow(<Icon rotation="90deg" {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the 90 deg counter-clockwise rotated style', () => {
    const wrapper = shallow(<Icon rotation="-90deg" {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the 180 deg rotated style', () => {
    const wrapper = shallow(<Icon rotation="180deg" {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the 180 deg counter-clockwise rotated style', () => {
    const wrapper = shallow(<Icon rotation="-180deg" {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the 90 deg rotate animated style', () => {
    const wrapper = shallow(<Icon rotation="90deg" smoothRotate {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the 90 deg counter-clockwise rotate animated style', () => {
    const wrapper = shallow(<Icon rotation="-90deg" smoothRotate {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the 180 deg rotate animated style', () => {
    const wrapper = shallow(<Icon rotation="180deg" smoothRotate {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the 180 deg counter-clockwise rotate animated style', () => {
    const wrapper = shallow(<Icon rotation="-180deg" smoothRotate {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the rotation resetted post-animation style', () => {
    const wrapper = shallow(<Icon rotation="reset" smoothRotate {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the 180 deg counter-clockwise rotate animated style', () => {
    const wrapper = shallow(<Icon rotation="-180deg" smoothRotate {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the bounce animation', () => {
    const wrapper = shallow(<Icon animation="bounce" {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the popIn animation', () => {
    const wrapper = shallow(<Icon animation="popIn" {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
