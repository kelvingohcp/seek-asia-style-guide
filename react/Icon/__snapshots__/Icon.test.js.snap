// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders with default props 1`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className=""
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon renders with title 1`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className=""
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      title
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the 90 deg counter-clockwise rotate animated style 1`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className="rotate90cc smoothRotate"
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the 90 deg counter-clockwise rotated style 1`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className="rotate90cc"
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the 90 deg rotate animated style 1`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className="rotate90 smoothRotate"
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the 90 deg rotated style 1`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className="rotate90"
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the 180 deg counter-clockwise rotate animated style 1`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className="rotate180cc smoothRotate"
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the 180 deg counter-clockwise rotate animated style 2`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className="rotate180cc smoothRotate"
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the 180 deg counter-clockwise rotated style 1`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className="rotate180cc"
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the 180 deg rotate animated style 1`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className="rotate180 smoothRotate"
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the 180 deg rotated style 1`] = `
<span
  className="root"
>
  <svg
    aria-hidden="true"
    className="rotate180"
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the bounce animation 1`] = `
<span
  className="root smallBounce"
>
  <svg
    aria-hidden="true"
    className=""
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the large style 1`] = `
<span
  className="root sizeLarge"
>
  <svg
    aria-hidden="true"
    className=""
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the popIn animation 1`] = `
<span
  className="root popIn"
>
  <svg
    aria-hidden="true"
    className=""
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;

exports[`Icon should render the small style 1`] = `
<span
  className="root sizeSmall"
>
  <svg
    aria-hidden="true"
    className=""
    focusable="false"
    role="img"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Chevron
    </title>
    <path
      d="M7.5 81.5l-6.3-6.7c-1.7-1.7-1.7-4.2 0-5.8L50 17.3 98.8 69c1.7 1.7 1.7 4.2 0 5.8l-6.3 6.7c-1.7 1.7-4.2 1.7-5.8 0L50 42.7 13.3 81.5c-1.6 1.6-4.1 1.6-5.8 0z"
      fill="currentColor"
      role="presentation"
    />
  </svg>
</span>
`;
